src/main/java/org/folio/linked/data/mapper/dto/monograph/common/place/OriginPlaceMapperUnit.java

```diff
@@ -0,0 +1,53 @@
+package org.folio.linked.data.mapper.dto.monograph.common.place;
+
+import static org.folio.ld.dictionary.PredicateDictionary.ORIGIN_PLACE;
+import static org.folio.ld.dictionary.ResourceTypeDictionary.PLACE;
+import static org.folio.linked.data.util.BibframeUtils.getFirstValue;
+import static org.folio.linked.data.util.Constants.IS_NOT_SUPPORTED_FOR_PREDICATE;
+import static org.folio.linked.data.util.Constants.RESOURCE_TYPE;
+import static org.folio.linked.data.util.Constants.RIGHT_SQUARE_BRACKET;
+
+import org.folio.linked.data.domain.dto.Place;
+import org.folio.linked.data.domain.dto.Work;
+import org.folio.linked.data.domain.dto.WorkReference;
+import org.folio.linked.data.exception.NotSupportedException;
+import org.folio.linked.data.mapper.dto.common.CoreMapper;
+import org.folio.linked.data.mapper.dto.common.MapperUnit;
+import org.folio.linked.data.model.entity.Resource;
+import org.folio.linked.data.service.HashService;
+import org.springframework.stereotype.Component;
+
+@Component
+@MapperUnit(type = PLACE, predicate = ORIGIN_PLACE, dtoClass = Place.class)
+public class OriginPlaceMapperUnit extends PlaceMapperUnit {
+
+  public OriginPlaceMapperUnit(CoreMapper coreMapper, HashService hashService) {
+    super(coreMapper, hashService);
+  }
+
+  @Override
+  public <P> P toDto(Resource source, P parentDto, Resource parentResource) {
+    if (parentDto instanceof Work || parentDto instanceof WorkReference) {
```

Comment:
Hi @askhat-abishev,

Below comment is for a future refactoring (next sprint may be)

I think we need to find a way to avoid these `instanceof` checks in our MapperUnits.

One Idea I have is to change the `SingleResourceMapperUnit` class as follows

```
class SingleResourceMapperUnit<P> {
   ...
}
```

Then change `InstanceSubResourceMapperUnit` as
```
public interface InstanceSubResourceMapperUnit extends SingleResourceMapperUnit<InstanceBase> {
 ...
}
```
(please read below to find what is InstanceBase)

This way, we can avoid the `instanceOf` checks in subclasses as below

```
public AccessLocationMapperUnit implements InstanceSubResourceMapperUnit { 
  public Instance toDto(Resource source, InstanceBase parentDto, Resource parentResource) {
     // note that the type of "parentDto" is "InstanceBase" here. So, no need to do "instanceof" check.
  }
}
```

Once challenge in doing this is the current `Instance`, `InstanceReference`, `Work`, `WorkReference` classes have no relation to each other today. The above given approach will work only if we can somehow make OpenAPI to generate a class hierarchy as follows

```
class ResourceBase {
  // common fields like Title goes here
}

class InstanceBase extends ResourceBase {
   // additional fields in Instance and InstanceRef goes here
}

class InstanceReference extends InstanceBase {
   // Most probably there won't be any additional properties here
}

class Instance extends InstanceBase {
  // additional properties in Instance goes here
}

class WorkBase extends ResourceBase {
   // additional fields in Instance and WorkRef goes here
}

class WorkReference extends WorkBase {
   // Most probably there won't be any additional properties here
}

class Work extends WorkBase {
  // additional properties in Work goes here
}
```

You are the experts. Can you please think about an option? @PBobylev  @aleksei-pronichev 

We can create a refactoring ticket for next sprint.

Reply:
Regarding OOP, it looks good.
Minor minuses minuses:
- Inheritance in DTO API objects is bad practice. For example, changing one field may change all the contracts in all endpoints. 
- While the code base is pretty less, support of it is more difficult

Alternative options for thinking how to refactor it:
- save logic, change check from `instance of` to
`if (Objects.equals (Work.class, parentDto.class)`

-- the same but more general
``` java
public static Set<Class<?>> workDtos - set.of(Work.class, WorkReference.class)
public static Set<Class<?>> instanceDtos - set.of(Instance.class, InstanceReference.class)
```
then
`if (intancesDtos.contains(parentDto.class))`

- next approach - think about factory (I'm not sure tha't it suitable
```
public interface mapperFactory implements Function<Class<?>, Mapper> // or BIFuncrion
```

Reply:
@pkjacob could you specify the approach that you want me to implement among those that mentioned in above comments?

Reply:
@askhat-abishev,
This looks good to me
```
//constanst
public static Set<Class<?>> workDtos - set.of(Work.class, WorkReference.class)
public static Set<Class<?>> instanceDtos - set.of(Instance.class, InstanceReference.class)

//then implement in class
`if (intancesDtos.contains(parentDto.class))
```

Reply:
decided to do the refactoring in a separate task